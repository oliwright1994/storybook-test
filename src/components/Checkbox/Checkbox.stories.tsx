import { Meta } from '@storybook/react/types-6-0'
import React from 'react'

import Checkbox, { ICheckbox } from './Checkbox'

export default { component: Checkbox, title: 'Components/Checkbox' } as Meta
export const Standard = (args: ICheckbox): JSX.Element => <Checkbox {...args} />
Standard.args = {
  label: 'Annual Management Charge',
  description: 'Charges will be disinvested annually',
  isDisabled: false,
  isInvalid: false,
  isRequired: false,
  isReadOnly: false,
  id: 'regulatory',
  value: 'regulatory_charge',
  errorMessage: 'You are not eligible for this charge configuration',
}
