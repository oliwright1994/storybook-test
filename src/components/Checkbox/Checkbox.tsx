import Stack from '@bright/components/Stack'
import { Space } from '@bright/constants'
import {
  Checkbox as ChakraCheckbox,
  FormControl,
  FormErrorMessage,
  FormHelperText,
  FormLabel,
  ICheckbox as ChakraICheckbox,
} from '@chakra-ui/core'
import React, { FC } from 'react'

interface ICheckbox
  extends Pick<ChakraICheckbox, 'isDisabled' | 'isInvalid' | 'isChecked' | 'isReadOnly' | 'onChange'> {
  id: string
  value: string | number
  label?: string
  errorMessage?: string
  description?: string
  variantColor?: string
  isRequired?: boolean
  size?: 'sm' | 'md' | 'lg'
}
const Checkbox: FC<ICheckbox> = ({
  id,
  label,
  description,
  variantColor,
  isChecked,
  onChange,
  isInvalid,
  isReadOnly,
  isDisabled,
  isRequired,
  errorMessage,
  size,
}) => {
  return (
    <FormControl isInvalid={isInvalid} isDisabled={isDisabled} isReadOnly={isReadOnly} isRequired={isRequired}>
      <Stack space={Space.xxs}>
        <ChakraCheckbox
          id={id}
          borderColor="gray.400"
          isInvalid={isInvalid}
          isDisabled={isDisabled}
          isReadOnly={isReadOnly}
          variantColor={variantColor}
          size={size}
          isChecked={isChecked}
          onChange={onChange}
          alignItems="baseline"
        >
          {label && (
            <FormLabel htmlFor={id} padding={0} verticalAlign="baseline">
              {label}
            </FormLabel>
          )}
          {description && <FormHelperText margin={0}>{description}</FormHelperText>}
        </ChakraCheckbox>
        {errorMessage && <FormErrorMessage>{errorMessage}</FormErrorMessage>}
      </Stack>
    </FormControl>
  )
}

export default Checkbox
