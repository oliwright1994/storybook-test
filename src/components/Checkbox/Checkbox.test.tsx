import { fireEvent } from '@testing-library/react'
import React from 'react'
import { renderWithBrightTheme } from 'testing/helpers/themeHelper'

import Checkbox from '.'

describe('<Checkbox />', () => {
  const defaultProps = {
    id: 'testCheckbox',
    value: 'testValue',
  }
  const label = 'This is a label'
  const description = 'This is a description'
  const errorMessage = 'This is an error message'

  it('should render as expected', () => {
    const { container } = renderWithBrightTheme(<Checkbox {...defaultProps} label={label} description={description} />)
    expect(container).toMatchSnapshot()
  })

  it('renders label', () => {
    const { getByText } = renderWithBrightTheme(<Checkbox {...defaultProps} label={label} />)
    expect(getByText(new RegExp(label, 'i'))).toBeTruthy()
  })

  it('renders description', () => {
    const { getByText } = renderWithBrightTheme(<Checkbox {...defaultProps} description={description} />)
    expect(getByText(new RegExp(description, 'i'))).toBeTruthy()
  })

  it('renders error message when isInvalid', () => {
    const { getByText } = renderWithBrightTheme(<Checkbox {...defaultProps} isInvalid errorMessage={errorMessage} />)
    expect(getByText(new RegExp(errorMessage, 'i'))).toBeTruthy()
  })

  it('has disabled attribute when passed isDisabled prop', () => {
    const { container } = renderWithBrightTheme(<Checkbox {...defaultProps} isDisabled />)
    expect(container.querySelector(`#${defaultProps.id}`)).toBeDisabled()
  })

  it('fires onChange={} ', () => {
    const handleChange = jest.fn()
    const { getByLabelText } = renderWithBrightTheme(
      <Checkbox {...defaultProps} label={label} onChange={handleChange} />
    )
    const checkboxElement = getByLabelText(label)
    expect(handleChange).toHaveBeenCalledTimes(0)
    fireEvent.click(checkboxElement)
    expect(handleChange).toHaveBeenCalledTimes(1)
  })
})
