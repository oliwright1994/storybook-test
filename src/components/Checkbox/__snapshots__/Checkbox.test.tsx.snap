// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> should render as expected 1`] = `
<div>
  <div
    class="css-0"
    role="group"
  >
    <div
      class="css-0"
    >
      <li
        class="css-1lxwves"
      >
        <label
          class="css-1j6qroy"
        >
          <input
            class="css-1gy22ay"
            id="testCheckbox"
            type="checkbox"
            value=""
          />
          <div
            aria-hidden="true"
            class="css-nlndso"
          >
            <svg
              class="css-1vhqk79"
              focusable="false"
              role="presentation"
              viewBox="0 0 14 14"
            >
              <g
                fill="currentColor"
              >
                <polygon
                  points="5.5 11.9993304 14 3.49933039 12.5 2 5.5 8.99933039 1.5 4.9968652 0 6.49933039"
                />
              </g>
            </svg>
          </div>
          <div
            class="css-6qsuox"
          >
            <label
              class="css-hny2kg"
              for="testCheckbox"
            >
              This is a label
            </label>
            <p
              class="css-1kvhzt2"
            >
              This is a description
            </p>
          </div>
        </label>
      </li>
      <li
        class="css-qzdso"
      />
    </div>
  </div>
</div>
`;
